UPGRADING.md

```diff
@@ -1,6 +1,44 @@
 Upgrading Hashie
 ================
 
+### Upgrading to 5.0.0
```

Comment:
I wonder whether we should do 4.1.0 for this. Some elements of the behavior change, but not major enough for API? @michaelherold what do you prefer?

Reply:
i was wavering on suggesting this...but it is not technically inconsistent with Semver to have entries in `changed` that don't correspond with a major version bump?

Reply:
What do I know? :) We've always treated the second digit as "major". We'd be at 24.0 otherwise and that feels like a lot. I'm open to this debate, but for now I say let's do 4.2.0. Will you make this change here @carolineartz, please?

Reply:
Actually .. re-reading what @michaelherold added to CHANGELOG I'll change my mind. Merging as is.

Reply:
Being strict with semver is always hard, especially when folks expect new features in the Major versions. I can agree with the bump to 5.0.0 if we've changed expected behavior in a way that could be breaking.

Reply:
We did. Amen.